src/cunumeric/sort/sort_omp.cc

```diff
@@ -0,0 +1,114 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/sort/sort.h"
+#include "cunumeric/sort/sort_template.inl"
+
+#include <thrust/sort.h>
+#include <thrust/execution_policy.h>
+#include <numeric>
+#include <omp.h>
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <LegateTypeCode CODE, int32_t DIM>
+struct SortImplBody<VariantKind::OMP, CODE, DIM> {
+  using VAL = legate_type_of<CODE>;
+
+  // sorts inptr in-place, if argptr not nullptr it returns sort indices
+  void thrust_local_sort_inplace(VAL* inptr,
+                                 int64_t* argptr,
+                                 const size_t volume,
+                                 const size_t sort_dim_size,
+                                 const bool stable_argsort)
+  {
+    if (argptr == nullptr) {
+      // sort (in place)
+#pragma omp parallel for
```

Comment:
I removed the manual pragmas and added the omp execution policy to the thrust call for now. This might not be optimal in all scenarios but keeps it simple until we decide to focus on it.